test/local/account_routes.js

```diff
@@ -392,6 +393,115 @@ test('/account/device', function (t) {
   })
 })
 
+test('/account/devices/notify', function (t) {
+  t.plan(4)
+  var config = {}
+  var uid = uuid.v4('binary')
+  var mockRequest = mocks.mockRequest({
+    credentials: {
+      uid: uid.toString('hex')
+    }
+  })
+  var pushPayload = {
+    isValid: true,
+    version: 1,
+    command: 'sync:collection_changed',
+    data: {
+      collections: ['clients']
+    }
+  }
+  var mockPush = mocks.mockPush()
+  var mockAjv = function () {
+    return {
+      compile: function () {
+        return function (payload) { return payload.isValid }
+      }
+    }
+  }
+  var accountRoutes = makeRoutes({
+    config: config,
+    push: mockPush
+  }, {
+    ajv: mockAjv
+  })
+  var route = getRoute(accountRoutes, '/account/devices/notify')
+
+  t.test('bad payload', function (t) {
+    mockRequest.payload = {
+      to: ['bogusid1'],
+      payload: {
+        isValid: false
+      }
+    }
+    return runTest(route, mockRequest, function () {
+      t.fail('should have thrown')
+    })
+    .catch(function (err) {
+      t.equal(mockPush.pushToDevices.callCount, 0, 'mockPush.pushToDevices was not called')
+      t.equal(err.errno, 107, 'Correct errno for invalid push payload')
```

Comment:
Can we also assert that the ajv validator function was called, to ensure that's actually what triggered the error?